### PR TITLE
Add inspect ownership command, add payee and signer to ocr2 inspection

### PR DESCRIPTION
**Pull request by kylesmartin**

### CODE REVIEW

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    static examples = [`yarn gauntlet ${contractId}:inspect_ownership --network=devnet [CONTRACT_ADDRESS]`]
+
+    constructor(flags, args) {
+      super(flags, args)
+    }
+
+    execute = async () => {
+      // Assert that only one argument was inputted
+      assertions.assert(this.args.length == 1, `Expected 1 argument, got ${this.args.length}`)
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts]: No need of this, we'll get the first element only anyway

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      // Get contract state
+      logger.info('Checking owner of ' + this.args[0])
+      const contract = getContract(contractId, '')
+      const program = this.loadProgram(contract.idl, contract.programId.toString())
+      const state = new PublicKey(this.args[0])
+      const onChainState = await program.account.state.fetch(state)
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts]: The way to fetch the state changes for each contract unfortunately:
- OCR: `program.account.state.fetch(state)`
- AccessController: `program.account.accessController.fetch(state)`
- Store: 
  - Store: `program.account.store.fetch(state)`
  - Transmissions: `program.account.transmissions.fetch(state)`

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      // Log owner of contract
+      const onChainOwner = onChainState.config.owner
+      logger.info('Owner is ' + onChainOwner)
+      // Return responses
+      return {
+        responses: [onChainOwner],
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts]: Let's include the data into `{ data: owner }`. Take a look at the `Result` structure

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/inspection/inspect.ts [hunk 4]
```diff
@@ -224,6 +253,20 @@ export default class OCR2Inspect extends SolanaCommand {
...
+      logger.info(
+        'Oracle ' +
+          oracle.transmitter +
+          '\n- Proposed Payee: ' +
+          oracle.proposedPayee +
+          '\n- From Round ID: ' +
+          oracle.fromRoundId +
+          '\n- Payment Gjuels: ' +
+          oracle.paymentGjuels +
+          '\n',
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/inspection/inspect.ts]: JS give us this string type which is more flexible
```suggestion
      logger.info(`Oracle Info:
          - Transmitter: oracle.transmitter
          - Proposed Payee: oracle.proposedPayee
          - From Round ID: oracle.fromRoundId
          - Payment Gjuels: oracle.paymentGjuels
          `)
```

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/inspection/inspect.ts [hunk 5]
```diff
@@ -65,6 +71,19 @@ export default class OCR2Inspect extends SolanaCommand {
     super(flags, args)
   }
 
+  // toHexString converts a list of numbers to a hex string
+  toHexString(nums: Int16Array): string {
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/inspection/inspect.ts]: `toHexString = (n: number[]) => Buffer.from(n).toString('hex')`

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts (hunk 2 above)
Comment (kylesmartin) [reply gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts]: Any suggestions for differentiating store vs transmission? I currently use a switch statement based on contractId so store and transmission contracts would have the same id. I could add in a new CONTRACT_LIST enum for this use case.

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    execute = async () => {
+      // Get contract owner
+      logger.info(`Checking owner of ${this.args[0]}`)
+      const contract = getContract(contractId, '')
+      const program = this.loadProgram(contract.idl, contract.programId.toString())
+      const state = new PublicKey(this.args[0])
+      const owner = await getOwner(program, state)
+      // Log owner of contract
+      logger.info(`Owner: ${owner}`)
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts]: Could we also print the `proposedOwner` if exists?

**Review by RodrigoAD — COMMENTED** [review]
Looks good, just a small request

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/accessController/index.ts [hunk 7]
```diff
@@ -12,12 +12,17 @@ const getOwner = async (program, state) => {
   return contractState?.owner
 }
 
+const getProposedOwner = async (program, state) => {
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/accessController/index.ts]: This is basically doing the same as `getOwner`. Let's refactor `getOwner` to `inspectOwnership` and return an `Ownership` type that is:
```js
type Ownership = {
  owner: string
  proposedOwner?: string
}
```
The `inspectOwnership` declaration should go where `makeInspectOwnershipCommand` is.

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { assertions, logger } from '@chainlink/gauntlet-core/dist/utils'
+import { SolanaCommand, TransactionResponse } from '@chainlink/gauntlet-solana'
+import { PublicKey } from '@solana/web3.js'
+import { CONTRACT_LIST, getContract } from '../../../lib/contracts'
+import { SolanaConstructor } from '../../../lib/types'
+
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts:8]:
```js
type InspectOwnership = (program: Program, account: PublicKey) => Promise<Ownership>
type Ownership = {
  owner: string,
  proposedOwner?: string
}
```

File: gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      // Get contract ownership
+      logger.info(`Checking owner of ${this.args[0]}`)
+      const contract = getContract(contractId, '')
+      const program = this.loadProgram(contract.idl, contract.programId.toString())
+      const state = new PublicKey(this.args[0])
+      const ownership = await getOwnership(program, state)
+      // Log owner of contract
+      logger.info(`Owner: ${ownership.owner}`)
+      // Log proposed owner of contract if it exists
+      if (ownership.proposedOwner!!) {
```
Comment (RodrigoAD) [inline gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ownership/inspectOwnership.ts]: Be careful here, `!` and the end of a variable tells TS that the variable cannot be null/undefined. So basically you are forcing it to pass this check always
```suggestion
      if (!!ownership.proposedOwner) {
```

**Review by RodrigoAD — COMMENTED** [review]
Just a final fix, we can merge after

**Review by RodrigoAD — APPROVED** [review]
Looks good! Thanks @kylesmartin

**Pull request merged**